### PR TITLE
[Merged by Bors] - chore(LinearAlgebra/QuadraticForm): fixups to #10097

### PR DESCRIPTION
**Pull request by eric-wieser**
I didn't get a chance to review #10097 before it was merged; this contains some minor fixups.

This removes `LinearMap.linMulLin`, as it can be recovered easily via `LinearMap.mul`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/QuadraticForm/Basic.lean [hunk 1]
```diff
@@ -663,8 +663,8 @@ section Semiring
 
 variable [CommSemiring R] [AddCommMonoid M] [Module R M]
 
-/-- A bilinear map into `R` gives a quadratic form by applying the argument twice. -/
-def _root_.LinearMap.toQuadraticForm (B: M →ₗ[R] M →ₗ[R] R) : QuadraticForm R M where
+/-- A bilinear map into `R` gives a quadratic form by applying the argument twice.  -/
```
Comment (Parcly-Taxel) [inline Mathlib/LinearAlgebra/QuadraticForm/Basic.lean:666]:
```suggestion
/-- A bilinear map into `R` gives a quadratic form by applying the argument twice. -/
```

**Review by Parcly-Taxel — APPROVED** [review]

**Review by mans0954 — APPROVED** [review]

File: Mathlib/Algebra/Algebra/Bilinear.lean [hunk 2]
```diff
@@ -22,6 +22,19 @@ open TensorProduct Module
...
+/-- A version of `LinearMap.restrictScalars` for bilinear maps. -/
+@[simps!]
+def restrictScalars₂
+    (R : Type*) {A M N P : Type*}
+    [CommSemiring R] [CommSemiring A]
+    [AddCommMonoid M] [AddCommMonoid N] [AddCommMonoid P]
+    [Algebra R A]
+    [Module R M] [Module R N] [Module R P]
+    [Module A M] [Module A N] [Module A P]
+    [IsScalarTower R A M] [IsScalarTower R A N] [IsScalarTower R A P]
```
Comment (jcommelin) [inline Mathlib/Algebra/Algebra/Bilinear.lean:34]: Could this share some variables with the rest of the file?

**Review by jcommelin — APPROVED** [review]
bors d+

File: Mathlib/Algebra/Algebra/Bilinear.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Algebra/Bilinear.lean:34]: I found doing so rather difficult, there are too many typeclasses I don't want earlier in the file, and confusion over linear vs semilinear maps.

**Comment by eric-wieser** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**